protocols/generic/generic.go

```diff
@@ -96,37 +127,114 @@ func build(tn *testnet.TestNet) error {
 		}
 
 		script := launchScript[node.GetRelativeNumber()]
+		buildParams := fmt.Sprintf("%v %s", script, params)
 
-		_, err = client.DockerExec(node, fmt.Sprintf("%v %s", script, params))
+		_, err = client.DockerExec(node, buildParams)
 		if err != nil {
 			return util.LogError(err)
 		}
 
 		tn.BuildState.IncrementBuildProgress()
 		return nil
 	})
-
-	return err
 }
 
-func idString(k crypto.PrivKey) string {
-	pid, err := peer.IDFromPrivateKey(k)
-	if err != nil {
-		panic(err)
+func createPeers(tn *testnet.TestNet, node ssh.Node, nodeKeyPairs map[string]crypto.PrivKey, networkTopology topology) (string, error) {
```

Comment:
ok, can you create a unit test for this method? I am not sure about the randomTwo case.

Also let's rename randomTwo to previousTwo, since they are no longer random.